src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -76,4 +94,22 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     Gson gson = new Gson();
     response.getWriter().println(gson.toJson(loginStatus));
   }
+
+  public String generateFriendCode() {
+    try {
+      String values = "0123456789abcdefghijklmnopqrstuvwxyz";
+      SecureRandom secureRandom = SecureRandom.getInstanceStrong();
+
+      // Generate a string of length 12.
+      String randomString =
+          secureRandom
+              .ints(12, 0, values.length())
+              .mapToObj(i -> values.charAt(i))
+              .collect(StringBuilder::new, StringBuilder::append, StringBuilder::append)
+              .toString();
+      return randomString;
+    } catch (NoSuchAlgorithmException e) {
```

Comment:
Returning an empty string and not logging the error is risky. I know this exception is extremely unlikely to happen, but if it does, it would silently create new users with no friend code. It would be better to fail the user creation and log an error.  Rather than adding it to this PR, perhaps create an issue and add a TODO to address this later.

Reply:
Done